### DISCUSSION TITLE
Searching with a large number causes a sql error

### DISCUSSION THREAD

**Discussion by appy-one** [directus/directus#15820] [Archive]
### Describe the Bug

Searching for items in the UI using a large number causes issues on `integer` data types in the collection: 

`2022-09-29 13:09:15.404 UTC [2113]: [13-1] db=hub,user=root ERROR: value "123456789901234560" is out of range for type integer`

### To Reproduce

* In a collection with an integer field, search for a large number such as `123456789901234560`.
* Inspect logs (if error not shown in UI)

### Errors Shown

`2022-09-29 13:09:15.404 UTC [2113]: [13-1] db=hub,user=root ERROR: value "123456789901234560" is out of range for type integer`

### What version of Directus are you using?

9.18.1

### What version of Node.js are you using?

16

### What database are you using?

Postgres

### What browser are you using?

Chrome / Firefox

### How are you deploying Directus?

Docker

**Comment by freekrai** [comment]
That’s actually a message coming from postgres telling you that particular number is indeed, out of range for an integer.

Can you try switching that field to a bigint instead?

**Comment by azrikahar** [comment]
> That’s actually a message coming from postgres telling you that particular number is indeed, out of range for an integer.

@freekrai is correct here. Perhaps a potential enhancement would be to make it so the App limits the max value that can be entered for integer fields, but technically this error will still (correctly) appear if you were to make such request directly via the API. 

Moved this to discussion for now, happy to explore any other alternatives if possible 👍

**Reply by keesvanbemmel** [reply]
Not quite sure why this issue was closed and converted into a discussion. If you create an integer field in directus and try to use the search endpoint, using a larger number than int allows, the API will crash. Of course this is caused by a database error, but since both the field creation/type setting and the search endpoint are in Directus, I strongly feel this can/should be an issue.

**Reply by azrikahar** [reply]
By crash, do you mean it no longer being responsive, or the same error returned as OP mentioned? I was not able to replicate the former definition of crash, but that would definitely be a more significant bug if that happens! Moving it to discussion does not mean it 100% cannot be an issue, but more so gather more info and/or evaluating it further, as we can always move it back to an issue afterwards 🙂

**Reply by keesvanbemmel** [reply]
We'll try to reproduce it and will let you know how!

**Comment by appy-one** [comment]
To clarify: the search is not done on an individual integer field of the collection, but on the collection itself. Directus executes a query that searches each field in the db table, causing the error.

**Reply by azrikahar** [reply]
Thanks for clarifying! That does make more sense now. I assume #14638 also had other integer fields in the same collection that wasn't brought up in the reproduction steps. 🤔

**Comment by FirstSanny** [comment]
Hello,

just wanna follow up on this one. We are having for example a database with products, which contain barcode fields. These are text fields, which can also contain EAN/GTIN codes. For weight and stuff we also have some integer fields on the collection.

But if I want to search for such a barcode now (which would be pretty normal behavior to search for a product by barcode) the UI is just prompting the Error, confusing actual users. 

I would propose to handle numbers larger than normal integers as strings/text instead. Shouldn't this prevent the error from getting thrown?
